src/actions-on-google.ts

```diff
@@ -351,4 +355,14 @@ export class ActionsOnGoogle {
             conversation.end()
         })
     }
-}
\ No newline at end of file
+}
+
+// Create some method aliases
+export interface ActionsOnGoogle {
```

Comment:
what are the pros and cons of aliasing this way versus just writing methods in the classes itself that calls what they alias?

Reply:
This discussion may help understand reason for aliasing - https://github.com/actions-on-google/actions-on-google-testing-nodejs/issues/8

Reply:
I mean about comparing aliasing using the prototype rather than explicitly writing methods in the class itself that just call the method they are aliasing.

```javascript
ClassName.prototype.alias = ClassName.prototype.original
```

compared to

```javascript
class ClassName {
  alias(args) {
    this.original(args)
  }
}
```

Reply:
@Canain I think that each approach has pros/cons:

* Add the alias to the prototype of the ActionsOnGoogle
  * Pros: The count of each alias method call doesn't increase.
  * Cons: Each definition of alias methods isn't placed in the ActionsOnGoogle class. Therefore, if using typedoc to generate API reference docs, the documents of the alias methods may not be generated.
* Define as an instance method of the ActionsOnGoogle
  * Pros: Each definition of alias methods is placed in the ActionsOnGoogle class. Therefore, if using typedoc to generate API reference docs, the documents of the alias methods can be generated as same as the original method.
  * Cons: The count of each alias method call increases one time ( `start()` -> `startConversation()` ).

If we should think about performance as high priority, we should select the approach of using the prototype. However, I guess that the performance is not high priority in this case. I wrote the code using the prototype, because Nick mentioned about the approach which defines alias methods without adding them as the method of the ActionsOnGoogle class at [#8](https://github.com/actions-on-google/actions-on-google-testing-nodejs/issues/8#issuecomment-403571926). But, maybe, it would be better that we select a simple approach rather than getting a little bit performance.

As the result, I intend to rewrite the code as a new method adding to the ActionsOnGoogle class. How about you?

Reply:
I think if we are going to support aliases, then explicitly declaring them as methods in the class is probably the better approach like you mentioned.

While it's more verbose, the declaration is more explicit and using standard OOP syntax meaning that it's more easily understood and parsable by tooling. The performance decrease shouldn't be important for a library like this.

Ideally, it would be nice to only support one name for when the library is stable, but it may not be feasible to solve for everyone.